api/pkg/db/model/model.go

```diff
@@ -35,13 +35,13 @@ type (
 
 	Catalog struct {
 		gorm.Model
-		Name       string
-		Type       string
-		URL        string
-		Owner      string
-		ContextDir string
+		Name       string `gorm:"unique_index:idx_name_org_code"`
+		Org        string `gorm:"unique_index:idx_name_org_code"`
+		Type       string `gorm:"not null;default:null"`
+		URL        string `gorm:"not null;default:null"`
+		ContextDir string `gorm:"not null;default:null"`
```

Comment:
I am not sure if we need "not null". Also why is the default: null when there is a constraint called `not null` 🤔

Reply:
`not null` is applied on table but when we create a record using `db.Create` if a field is not mentioned in struct, gorm adds 
`""` by default.
 So, for example if `url` is not defined then ideally it should return error but as gorm add `""`, it does not return error. due to which we add default as null and expects some value.